stix_shifter_modules/crowdstrike/stix_transmission/connector.py

```diff
@@ -31,45 +36,76 @@ def _handle_errors(self, response, return_obj):
-            raise Exception(return_obj)
-        elif ErrorResponder.is_json_string(response_txt):
-            response_json = json.loads(response_txt)
-            ErrorResponder.fill_error(return_obj, response_json, ['reason'], connector=self.connector)
-            raise Exception(return_obj)
-        else:
-            raise Exception(return_obj)
+        elif response_code >= 400:
+            if response_type == 'application/json':
+                error_response = json.loads(response_txt)
+                response_dict['type'] = 'ValidationError'
+                response_dict['message'] = error_response['errors'][0]['message']
+                ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+                raise QueryException(return_obj)
+            elif response_type == 'text/html':
+                error = ConnectionError(f'Error connecting the datasource: {response_txt}')
+                ErrorResponder.fill_error(return_obj, response_dict, error=error, connector=self.connector)
+                raise QueryException(return_obj)
+            else:
+                raise Exception(response_txt)
 
+    
     async def ping_connection(self):
         response_txt = None
         return_obj = {}
+        response_dict = {}
         try:
             response = await self.api_client.ping_box()
             response_code = response.code
             response_txt = response.read().decode('utf-8')
+            response_type = response.headers.get('Content-Type')
             if 199 < response_code < 300:
                 return_obj['success'] = True
-            elif isinstance(json.loads(response_txt), dict):
-                response_error_ping = json.loads(response_txt)
-                response_dict = response_error_ping['errors'][0]
-                ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+            elif response_code == 401:
+                if response_type == 'application/json':
+                    error_response = json.loads(response_txt)
+                    response_dict['type'] = 'AuthenticationError'
+                    response_dict['message'] = error_response['errors'][0]['message']
+                    self.logger.error('Error connecting the datasource: ' + str(error_response))
+                    ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+                else:
+                    raise Exception(response_txt)
+            elif response_code == 400:
+                if response_type == 'application/json':
+                    error_response = json.loads(response_txt)
+                    response_dict['type'] = 'ValidationError'
+                    response_dict['message'] = error_response['errors'][0]['message']
+                    self.logger.error('Error connecting the datasource: ' + str(error_response))
```

Comment:
Do we need to add the connector name here?

Reply:
yeah we can add connector name here in error logger.